### ISSUE TITLE
parser: syntax error: Token 'data' is unexpected here

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#3092]
```fortran
module BCD_TO_ASCII_M                         
                                                      
  implicit NONE                                
  public                                      
                                               
  integer :: ASCII_TO_BCD(0:255) 
  data ASCII_TO_BCD / &                     
  ! 0000 - 0037: Control chars        
    o'00', o'00', o'00', o'00', o'00', o'00', o'00', o'00', &
    o'00', o'00', o'00', o'00', o'00', o'00', o'00', o'00', &
    o'00', o'00', o'00', o'00', o'00', o'00', o'00', o'00', &
...
```
gives:
```console
$ lfortran bcd_to_ascii_m.f90 
syntax error: Token 'data' is unexpected here
 --> bcd_to_ascii_m.f90:7:3
  |
7 |   data ASCII_TO_BCD / &
  |   ^^^^ 
```

**Comment by pjh40** [comment]
This doesn't seem like an issue any more, compiles without warning in main.

**Comment by certik** [comment]
Thanks for noticing! I used git bisection, this was fixed by be475026bd0becde3097dd00af6dde35224c53a6 from #4667. This issue is a duplicate of #4666. A test was added in `integration_tests/data_12.f90`, so we can close this now.
